### PR TITLE
Properly free CompilationInfoPerThread resources

### PR DESCRIPTION
**Pull request by a7ehuo**
Properly free CompilationInfoPerThread resources

Port the change #7655 from the `jitaas` branch to the `master` branch. 

Fixes: #7338

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@mpirvu Ready for review. Thanks!

**Comment by mpirvu** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by mpirvu** [comment]
Jenkins test sanity aix jdk8,jdk11

**Comment by mpirvu** [comment]
The AIX tests failed due to infra. This PR is good to be merged.

File: runtime/compiler/control/JITServerCompilationThread.cpp [hunk 1]
```diff
@@ -1039,3 +1039,14 @@ TR::CompilationInfoPerThreadRemote::deleteClientSessionData(uint64_t clientId, T
       }
    compInfo->releaseCompMonitor(compThread);
    }
+
+void
+TR::CompilationInfoPerThreadRemote::freeAllResources()
+   {
+   if (_recompilationMethodInfo)
+      {
+      TR_Memory::jitPersistentFree(_recompilationMethodInfo);
```
[the remark this answers is not in the document]
Comment (a7ehuo) [reply runtime/compiler/control/JITServerCompilationThread.cpp]: Created #7988 to track this follow up item.

**Pull request merged**